src/bun.js/bindings/helpers.h

```diff
@@ -79,6 +79,11 @@ static const WTF::String toString(ZigString str)
     }
     if (isTaggedUTF8Ptr(str.ptr)) [[unlikely]] {
         ASSERT_WITH_MESSAGE(!isTaggedExternalPtr(str.ptr), "UTF8 and external ptr are mutually exclusive. The external will never be freed.");
+        // This will fail if the string is too long. Let's make it explicit instead of an ASSERT.
+        // Check both the Bun synthetic limit and WebKit's String::MaxLength.
+        if (str.len > Bun__stringSyntheticAllocationLimit || str.len > WTF::String::MaxLength) [[unlikely]] {
+            return {};
```

Comment:
Can you use SIMDUTF's length-checking functions to verify the length is too big when inside this branch before we return {}?